lib/greed/roll_score.rb

```diff
@@ -0,0 +1,62 @@
+module Greed
+  class RollScore
+    attr_reader :value
+
+    def initialize(dice)
+      raise ArgumentError, "Valid roll object not provided" unless dice
+      @dice = dice
+      @value = calculate_score
+    end
+
+    def non_scoring_dice
```

Comment:
I don't like this method for several reasons

It does too much
It does things it shouldn't do.

A RollScore object should not know about returning a new set of dice if there are no dice that are not involved in scoring and it should not be instantiating new die objects itself.  That seems more like it should belong to a Roll knowing that at the end of a turn it should have some amount of dice not involved in scoring or somewhere else.

The logic of if a die / group of dice does contribute to score is something that I would like to see in a private method and not just jammed in here personally since going by the name of the function it should just return `non_scoring_dice` not decide itself if a group of dice is scoring or not.